src/components/tips/tip-uploader.tsx

```diff
@@ -0,0 +1,301 @@
+        duration: 3000,
+        icon: '✅',
+      })
+      form.reset()
+      sets3Url('')
+      form.setValue('s3Url', '')
+
+      await trpcUtils.tips.invalidate()
+      router.refresh()
+    },
+    onError: (error) => {
+      toast.error(
+        `There was a problem creating this tip. Contact egghead staff if the issue persists.`,
+        {
+          duration: 3000,
+          icon: '❌',
+        },
+      )
+    },
+  })
+
+  const onSubmit = async (values: z.infer<typeof NewTipFormSchema>) => {
+    createTip({
+      ...values,
+      fileName: slugify(values.title.toLowerCase(), {remove: /[*+~.()'"!:@]/g}),
+    })
+  }
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-8">
+        <FormField
+          control={form.control}
+          name="title"
+          render={({field}) => (
+            <FormItem>
+              <FormLabel className="text-lg font-bold">Title*</FormLabel>
+              <FormDescription className="mt-2 text-sm">
+                Include any relavent keywords for SEO.
+              </FormDescription>
+              <FormControl>
+                <Textarea placeholder="How do I..." {...field} />
+              </FormControl>
+
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={form.control}
+          name="description"
+          render={({field}) => (
+            <FormItem>
+              <FormLabel className="text-lg font-bold">Description</FormLabel>
+              <FormDescription className="mt-2 text-sm">
+                This should describe what you teach in the tip and include any
+                relevant codeblocks and links.
+              </FormDescription>
+              <FormControl>
+                <Textarea
+                  placeholder="add as much detail as you'd like :)"
+                  {...field}
+                />
+              </FormControl>
+
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+
+        {s3Url.length === 0 ? (
+          <TipUploader
+            sets3Url={(s3Url: string) => {
+              form.setValue('s3Url', s3Url)
+              sets3Url(s3Url)
+            }}
+          />
+        ) : (
+          <div>
+            {s3Url}
+            <Button
+              onClick={() => {
+                sets3Url('')
+                form.setValue('s3Url', '')
+              }}
+            >
+              clear
+            </Button>
+          </div>
+        )}
+        <FormField
+          control={form.control}
+          name="s3Url"
+          render={({field}) => (
+            <FormItem>
+              <FormControl>
+                <Input type="hidden" {...field} />
+              </FormControl>
+              {form.formState.errors.s3Url && (
+                <p>{form.formState.errors.s3Url.message}</p>
+              )}
+            </FormItem>
+          )}
+        />
```

Comment:
the error here says you need atleast 4 characters which is confusing, need to update with a custom error message that says please upload a video

Reply:
`z.string().trim().url({ message: 'Please Upload a Video' })`

added this to the schema